### ISSUE TITLE
Problem with cli examples

### ISSUE THREAD

**Issue by blackshirt** [vlang/v#8699]
i have fun playing with vlang cli module.
for first try, i'm looking from cli examples [https://github.com/vlang/v/blob/master/examples/cli.v](url), but i have a problem how i  should have to pass the right flag to the examples.
here the output i'm trying to pass
```
green@ubuntu:~/vexper/cmd$ v run main.v greet
Command `greet` needs at least 1 arguments
green@ubuntu:~/vexper/cmd$ v run main.v greet 'blah'
This is a function running before the main function.

Unsupported language
Supported languages are `english`, `german` and `dutch`.
cli execution error: ���
green@ubuntu:~/vexper/cmd$ v run main.v greet --language='english'
Command `greet` needs at least 1 arguments
green@ubuntu:~/vexper/cmd$ v run main.v greet --language='english' 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet language='english' 'blah'
This is a function running before the main function.

Unsupported language
Supported languages are `english`, `german` and `dutch`.
cli execution error: �6>L�
green@ubuntu:~/vexper/cmd$ v run main.v greet language 'english' 'blah'
This is a function running before the main function.

Unsupported language
Supported languages are `english`, `german` and `dutch`.
cli execution error: ����
green@ubuntu:~/vexper/cmd$ v run main.v greet -l 'english' 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet -l='english' 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet -l='english' --times=3 --fun=fun 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet -l=english --times=3 --fun=fun 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet -l 'en' --times 3 --fun='fun' 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$ v run main.v greet -l 'en' --times 3 --fun 'fun' 'blah'
Flag `language` is required by `cli greet`
green@ubuntu:~/vexper/cmd$
```

what's happens really, i think i'm passing the right value for required flag.
Any help for this silly issue ??

Thanks, 
blackshirt

**Comment by cbracketdash** [comment]
The name of the file is cli.v ...

**Comment by blackshirt** [comment]
the point is not about cli.v or main.v (main.v contents exactly same as cli.v).
commonly passing flag was in the form
```
$command --flag='somevalue' --secondflag='secondvalue' 
```
or sometimes 
```
$command --flag 'somevalue' --secondflag 'secondvalue' 
```
using short abbreviation was a valid form.
if we look at [https://github.com/vlang/v/blob/master/examples/cli.v](url), `language` flag was marked as required flag.
Even i passing the flag, cli always complaints ```Flag `language` is required by `cli greet```` 

I think there are subtle bug in the flag handling of this (but, imho, just my think, i dont look deep a bit to the source)

What do you think about it?

**Comment by spytheman** [comment]
The point of the remark about the name, is that you are making it, needlessly harder for people to reproduce your problem @blackshirt . Making the example you are giving in an issue as accurate as possible, is a sign of respect to the people that will try to help you, and will increase the chances of your problem being solved.

It is fixed now in #8826 .

**Comment by blackshirt** [comment]
Thanks @spytheman for the suggestions, i would try the best 

btw, good works
